cupy/core/fusion.py

```diff
@@ -737,13 +738,16 @@ def __repr__(self):
         return '<Fusion \'{}\'>'.format(self.name)
 
     def __call__(self, *args, **kwargs):
-        _thread_local.in_fusion = True
-        _thread_local.history = _FusionHistory()
-        try:
-            return self._call(*args, **kwargs)
-        finally:
-            _thread_local.in_fusion = False
-            del _thread_local.history
+        if _thread_local.in_fusion:
+            return self.func(*args, **kwargs)
+        else:
+            _thread_local.in_fusion = True
+            _thread_local.history = _FusionHistory()
```

Comment:
How about integration `in_fusion` and `history` ?

Reply:
Fixed.